Solaris/DmarcAggregateParser.php

```diff
@@ -120,6 +122,12 @@ function parse( $files ) {
 			}
 		}
 
+		if (empty($this->errors)) {
```

Comment:
This logic doesn't look right to me. A few comments:

- Database-level exceptions will only be raised when you actually try to commit the changes (which you are doing inside this block) and will not be caught.

- `errors` will no longer be populated with any exception messages.

- Also if you look at the cases in which `$this->errors` are populated above, not all of them warrant a rollback (e.g., ignoring repeated reports).

I think it may be easier to perform each item in the record loop as a transaction, instead of the whole import. That would still address your issue of failed child imports. The `commit()` needs to be wrapped in a try/catch block and the exceptions logged to `errors`.